include/nbl/asset/ICPUPolygonGeometry.h

```diff
@@ -0,0 +1,176 @@
+// Copyright (C) 2025-2025 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_ASSET_I_CPU_POLYGON_GEOMETRY_H_INCLUDED_
+#define _NBL_ASSET_I_CPU_POLYGON_GEOMETRY_H_INCLUDED_
+
+
+#include "nbl/asset/IPolygonGeometry.h"
+
+
+namespace nbl::asset
+{
+//
+class NBL_API2 ICPUPolygonGeometry : public IAsset, public IPolygonGeometry<ICPUBuffer>
+{
+        using base_t = IPolygonGeometry<ICPUBuffer>;
+
+    public:
+        inline ICPUPolygonGeometry() = default;
+        
+        constexpr static inline auto AssetType = ET_GEOMETRY;
+        inline E_TYPE getAssetType() const override {return AssetType;}
+
+        inline core::smart_refctd_ptr<IAsset> clone(uint32_t _depth=~0u) const
+        {
+            const auto nextDepth = _depth ? (_depth-1):0;
+            auto retval = core::smart_refctd_ptr<ICPUPolygonGeometry>();
+            retval->m_positionView = m_positionView.clone(nextDepth);
+            retval->m_jointOBBView = m_jointOBBView.clone(nextDepth);
+            retval->m_indexView = m_indexView.clone(nextDepth);
+            retval->m_jointWeightViews.reserve(m_jointWeightViews.size());
+            for (const auto& pair : m_jointWeightViews)
+                retval->m_jointWeightViews.push_back({
+                    .indices = pair.indices.clone(nextDepth),
+                    .weights = pair.weights.clone(nextDepth)
+                });
+            retval->m_auxAttributeViews.reserve(m_auxAttributeViews.size());
+            for (const auto& view : m_auxAttributeViews)
+                retval->m_auxAttributeViews.push_back(view.clone(nextDepth));
+            retval->m_normalView = m_normalView.clone(nextDepth);
+            retval->m_jointCount = m_jointCount;
+            retval->m_verticesForFirst = m_verticesForFirst;
+            retval->m_verticesPerSupplementary = m_verticesPerSupplementary;
+            return retval;
+        }
+
+        // TODO: remove after https://github.com/Devsh-Graphics-Programming/Nabla/pull/871 merge
+        inline size_t getDependantCount() const override
+        {
+           size_t count = 0;
+           visitDependents([&current](const IAsset* dep)->bool
+               {
+                   count++;
+                   return true;
+               }
+           );
+           return count;
+        }
```

Comment:
@kevyuu this is what I had in mind for the visitor pattern & callback https://github.com/Devsh-Graphics-Programming/Nabla/pull/871